Consistently use `vega-util` error instead of `throw new Error(...)`

Like in https://github.com/vega/vega-lite/pull/3394

When I try to throw the error using error function from `vega-util`, the `tsc` is giving the error (`error TS7030: Not all code paths return a value.`) which is possibly due to the fact that no return value is explicitly specified when throwing the error by calling a function.

--- a/src/compile/projection/assemble.ts
+++ b/src/compile/projection/assemble.ts
@@ -1,4 +1,3 @@
-import {error} from 'vega-util';
 import {contains} from '../../util';
 import {isVgSignalRef, VgProjection, VgSignalRef} from '../../vega.schema';
 import {isConcatModel, isLayerModel, isRepeatModel, Model} from '../model';
@@ -40,7 +39,7 @@ export function assembleProjectionForModel(model: Model): VgProjection[] {
   }, []);
 
   if (fit.length <= 0) {
-    error("Projection's fit didn't find any data sources");
+    throw new Error("Projection's fit didn't find any data sources");
   }
 
   return [{

